apps/website/src/server/context.ts

```diff
@@ -0,0 +1,28 @@
+import { loginPresets } from '@bluedot/ui';
+import * as trpcNext from '@trpc/server/adapters/next';
+
+export const createContext = async ({ req }: trpcNext.CreateNextContextOptions) => {
+  const authHeader = req.headers.authorization;
+
+  // Only attempt to verify if we have a valid Bearer token format
+  if (!authHeader || !authHeader.startsWith('Bearer ')) {
+    return { auth: null };
+  }
+
+  const token = authHeader.slice('Bearer '.length).trim();
+
+  try {
+    const auth = await loginPresets.keycloak.verifyAndDecodeToken(token);
+    return { auth };
+  } catch (error) {
+    // TODO: do we need this fallback check to googleBlueDot?
+    // try {
+    //   auth = await loginPresets.googleBlueDot.verifyAndDecodeToken(token);
+    // } catch {
+    // }
```

Comment:
I'm not sure how to handle this, or what the strategy regarding two login presets should be? Should we just try to authenticate both, and only if both fail do we set auth to null?